### PR TITLE
Add PointerValue.Value to dereference and access pointer value.

### PR DESCRIPTION
**Pull request by joncham**
I was not sure about:

1. Protocol versioning process
2. CommandSet enum value

### CODE REVIEW

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 1]
```diff
@@ -4412,5 +4414,35 @@ public void Bug59649 ()
 		// DummyCall
 		assert_location (e, "Call");
 	}
+
+	[Test]
+	public void Pointer_GetValue () {
+		var e = run_until ("pointer_arguments");
+		var frame = e.Thread.GetFrames () [0];
+
+		var param = frame.Method.GetParameters()[0];
```
Comment (joncham) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Looks like I have some formatting issues.

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs [hunk 2]
```diff
@@ -845,9 +850,12 @@ class PacketReader {
 					return new ValueImpl { Type = etype, Value = ReadDouble () };
 				case ElementType.I:
 				case ElementType.U:
-				case ElementType.Ptr:
 					// FIXME: The client and the debuggee might have different word sizes
 					return new ValueImpl { Type = etype, Value = ReadLong () };
+				case ElementType.Ptr:
+					long value = ReadLong();
+					long pointerClass = ReadId();
+					return new ValueImpl {Type = etype, Klass = pointerClass, Value = value};
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs]: This is a breaking protocol change, so it needs to be inside a:
								if (Version.AtLeast (2, 46))

**Comment by jbevain** [comment]
Related: https://github.com/mono/monodevelop/issues/3673

**Comment by migueldeicaza** [comment]
One thing that we could do is to notify the runtime that we are about to poke into no-mans-land with disastrous effects, to prevent the native exception handler from running.

Something that would inform `mono_handle_native_crash` to not dump the stack, but instead set a flag, and then communciate the failure to the client.

File: mono/mini/debugger-agent.c [hunk 3]
```diff
@@ -9991,6 +9998,34 @@ string_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 	return ERR_NONE;
 }
 
+static ErrorCode
+pointer_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
+{
```
Comment (vargaz) [inline mono/mini/debugger-agent.c]: command_set_to_string () etc needs to be modified to handle the new command set/commands.

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs (hunk 2 above)
Comment (joncham) [reply mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs]: @vargaz this is inside the PacketReader class that doesn't have access to connection/vm to check version. What is best way to handle this?

Comment (vargaz) [reply mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs]: Pass the connection instance to the PacketReader ctor.

**Comment by marek-safar** [comment]
@jchannon could you rebase it and address @vargaz review

**Comment by jchannon** [comment]
Think you mean @joncham

**Comment by marek-safar** [comment]
@jchannon absolutely, sorry for the typo

**Comment by jchannon** [comment]
mp

**Comment by joncham** [comment]
Working on it today

**Comment by joncham** [comment]
Responded to comments. Added `TypeMirror.GetValueSize` and test for basic pointer arithmetic support. IDE devs ( @jbevain ) thoughts?

**Pull request merged**